projects/packages/protect-status/src/class-scan-status.php

```diff
@@ -223,24 +222,38 @@ private static function normalize_api_data( $scan_data ) {
 					$status->fixable_threat_ids[] = $scan_threat->id;
 				}
 
+				$db_details = null;
+				if ( ! empty( $scan_threat->table ) ) {
+					$db_details = array_merge(
+						array(
+							'table'     => $scan_threat->table,
+							'pk_column' => $scan_threat->pk_column ?? null,
+							'pk_value'  => $scan_threat->value ?? null,
+						),
+						! empty( $scan_threat->details ) ? array( 'details' => $scan_threat->details ) : array()
+					);
```

Comment:
```suggestion
							'pk_value'  => $scan_threat->value ?? null,
							'details'   => $scan_threat->details ?? array(),
						),
					);
```

Why not simplify that a bit. And remove the `array_merge()` call.

Reply:
Well, it does not do the same thing. Currently, we don't always have the `details` index, with this change we would always have it, and it would be an empty array if there are none. Also, `! empty()` is not the same check as the null-coalescing operator (though that might not matter too much here).

I'll go and see if it works just as well with the simplification.

Reply:
So about this, it would currently show the details as `null` which I think is not overly helpful. 

![image](https://github.com/user-attachments/assets/07845851-e8ca-496a-a83e-0f5f3e3d4941)

I could technically change the `details-viewer` component to not display null-values, but then again, maybe sometimes you want to. I'd say let's leave it like it is.